step-automation-packages/step-automation-packages-client/src/main/java/step/automation/packages/client/RemoteAutomationPackageClientImpl.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.automation.packages.client;
+
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.Invocation;
+import jakarta.ws.rs.core.MediaType;
+import org.glassfish.jersey.media.multipart.MultiPart;
+import org.glassfish.jersey.media.multipart.file.FileDataBodyPart;
+import step.client.AbstractRemoteClient;
+import step.client.credentials.ControllerCredentials;
+
+import java.io.File;
+import java.util.function.Function;
+
+public class RemoteAutomationPackageClientImpl extends AbstractRemoteClient implements AutomationPackageClient {
+
+    public RemoteAutomationPackageClientImpl() {
+        super();
+    }
+
+    public RemoteAutomationPackageClientImpl(ControllerCredentials credentials) {
+        super(credentials);
+    }
+
+    @Override
+    public String newAutomationPackage(File automationPackageFile) {
+        return uploadPackage(automationPackageFile, multiPartEntity -> {
+            Invocation.Builder builder = requestBuilder("/rest/automation-packages");
+            return RemoteAutomationPackageClientImpl.this.executeRequest(() -> builder.post(multiPartEntity, String.class));
+        });
+    }
+
+    @Override
+    public String createOrUpdateAutomationPackage(File automationPackageFile) {
+        return uploadPackage(automationPackageFile, multiPartEntity -> {
+            Invocation.Builder builder = requestBuilder("/rest/automation-packages");
```

Comment:
@iegorov777 The put service should have the package name in URI, should it be done as part of this CR?

Reply:
@david-stephan
I thought about that, but there is a difficulty, because to pass the package name in URI we should either add this name to plugin configuration (and this name should be the same as in automation descriptor). 
Or we should extract the name from automation package file (unzip), which also doesn't make sense...